### PR TITLE
[RW-1416][RW-1355][risk=low] Rename CDRs, fix prod CDR config

### PR DESCRIPTION
**Pull request by calbach**
- Fix prod CDR configs to point to an actual CDR that's available in prod (it's currently broken)
- Rename to "Dataset" to avoid the opaque "CDR" acronym
- Require an "id" in CDR JSONs; use this as the primary key
-> I prefer this to introducing a new ID type

### CODE REVIEW

**Comment by danqrodney** [comment]
I have no objections to the naming convention.

We will need to hack our database if we want to do a rename of the CDR version without breaking things, though.

**Comment by danqrodney** [comment]
Note: I could imagine us adding some other external ID to CDR versions so that we can rename them in future without hacking the DB.

**Comment by calbach** [comment]
When I sent this PR that was the plan, but since we got feedback during the demo it sounded like that was less solid. In any case I'm now tracking this via https://precisionmedicineinitiative.atlassian.net/browse/RW-1355

File: api/config/cdr_versions_stable.json [hunk 1]
```diff
@@ -11,7 +11,7 @@
     "publicDbName": "public20180831"
   },
   {
-    "name": "Test Registered CDR 20180920",
+    "name": "Synthetic Dataset v2",
```
Comment (calbach) [inline api/config/cdr_versions_stable.json]: @danqrodney this one isn't "releaseNumber=2", should it be? I don't have context on this field.

Comment (danqrodney) [reply api/config/cdr_versions_stable.json]: Release number is there as an ordering of releases; in future we expect to have both registered and controlled CDR versions with the same release number, released around the same time. For now it doesn't mean a whole lot.

Comment (calbach) [reply api/config/cdr_versions_stable.json]: Got it. Is there a reason test is different from local/staging/stable here? (test sets releaseNumber=2)

File: api/config/cdr_versions_prod.json [hunk 2]
```diff
@@ -1,13 +1,13 @@
...
+    "name": "Synthetic Dataset v1",
     "dataAccessLevel": 1,
     "bigqueryProject": "all-of-us-ehr-dev",
     "bigqueryDataset": "test_merge_dec26",
     "creationTime": "2017-12-26 00:00:00Z",
     "releaseNumber": 1,
     "numParticipants": 946237,
-    "cdrDbName": "cdr20180831",
-    "publicDbName": "public20180831"
+    "cdrDbName": "cdr20180817",
```
Comment (calbach) [inline api/config/cdr_versions_prod.json]: 20180831 doesn't exist in prod

**Comment by calbach** [comment]
Updated: I deviated from your recommendation by just using the primary DB ID here. I don't see a good reason to introduce another ID concept since we're effectively just mirroring this JSON in MySQL anyways. Let me know if you have opinions on this.

**Pull request merged**